pkg/segment/aggregations/streamstats.go

```diff
@@ -392,25 +452,42 @@ func performMeasureFunc(currIndex int, ssResults *structs.RunningStreamStatsResu
 			return 0.0, fmt.Errorf("performMeasureFunc: Error while getting float value from min window element, err: %v", err)
 		}
 		ssResults.CurrResult = maxFloatVal - minFloatval
+	case utils.Cardinality, utils.Values:
+		if ssResults.CardinalityMap == nil {
+			ssResults.CardinalityMap = make(map[string]int, 0)
+		}
+		strValue := fmt.Sprintf("%v", colValue)
+		_, exist := ssResults.CardinalityMap[strValue]
+		if !exist {
+			ssResults.CardinalityMap[strValue] = 1
+		} else {
+			ssResults.CardinalityMap[strValue]++
+		}
```

Comment:
fyi: you can replace this block with just `ssResults.CardinalityMap[strValue]++` (see https://go.dev/play/p/CAHSqqanSgu)

However, what you have is quite readable, so feel free to keep it

Reply:
I think it would be better to keep it for readability.